File: armstrongNumber.java
import java.util.*;
class ArmstrongNumber{
public static void main(String[] args){
        int a=153;
        boolean ans=isArmstrong(a);
       
        System.out.println(ans);
}
public static boolean isArmstrong(int n){
    int cnt=0;
    int temp=n;
    while(temp!=0){
        cnt++;
        temp=temp/10;

    }
    int m=n;
    int rv=0;
    while(m!=0){
        int dig=m%10;
        rv+=Math.pow(dig,cnt);
        m=m/10;
        
    }
    if(rv==n){
        return true;
    }
    return false;

}
}

File: armstrongNumber.js
// const isArm = (n)=>{
//     let temp=n;
//     let sum=0;
//     let k=String(n).length;
//     while(temp!==0){
//         let dig=temp%10;
//         sum+=Math.pow(dig,k);
//         temp=Math.trunc(temp/10);
//     }
//     if(n===sum){
//         return true;
//     }
//     return false;

// }
// let n=153;
// console.log(isArm(n));


// const isA=(n)=>{
//     let cnt=0;
//     let temp=n;
//     while(temp!=0){
//         cnt++;
//         temp=Math.trunc(temp/10);
//     }
//     let m=n;
//     let ans=0;
//     while(m!=0){
//         let dig=m%10;
//         ans=ans+Math.pow(dig,cnt);
//         m=Math.trunc(m/10);


//     }
//     if(ans==n){
//         return true;
//     }
//     return false;
// }

// let n=153;
// console.log(isA(n));